R/DataMart.R

```diff
@@ -483,3 +483,221 @@ IsDisplayrCloudDriveAvailable <- function()
     company.secret <- get0("companySecret")
     !is.null(company.secret) && company.secret != "UNKNOWN"
 }
+
+uploadRScript <- function(r.code,
+                          js.code = NULL,
+                          filename,
+                          upload = TRUE,
+                          api.root = Sys.getenv("API_ROOT"),
+                          company.secret = Sys.getenv("COMPANY_SECRET"),
+                          client.id = Sys.getenv("CLIENT_ID"))
+{
+    if (missing(filename))
+        stop(substitute(filename), " argument required as ",
+             "filename to write in Displayr Drive is required.")
+    type <- file_ext(filename)
+    stopifnot("rscript filename extension required" = type == "rscript")
+    if (upload)
+        checkUploadPossible(api.root = api.root, company.secret = company.secret, client.id = client.id)
+    tmpfile <- tempfile()
+    file <- file(tmpfile, "wb")
+    r.filenames <- if (is.list(r.code)) unlist(list(r.code)) else r.code
```

Comment:
why if `is.list` is `TRUE` do you then call `list()` again?

Reply:
Oops, that internal `list` should be removed. I'll fix.

Reply:
Fixed in https://github.com/Displayr/flipAPI/pull/12/commits/d7c16db9c43cd6440ac24331713af88ad4f236cd